utilities/rpp-unittests/SOW3_HOST/BatchPD_host_pln3.cpp

```diff
@@ -0,0 +1,921 @@
+        srcSize[count].height = image.rows;
+        srcSize[count].width = image.cols;
+        if (maxHeight < srcSize[count].height)
+            maxHeight = srcSize[count].height;
+        if (maxWidth < srcSize[count].width)
+            maxWidth = srcSize[count].width;
+        if (minHeight > srcSize[count].height)
+            minHeight = srcSize[count].height;
+        if (minWidth > srcSize[count].width)
+            minWidth = srcSize[count].width;
+
+        dstSize[count].height = image.rows;
+        dstSize[count].width = image.cols;
+        if (maxDstHeight < dstSize[count].height)
+            maxDstHeight = dstSize[count].height;
+        if (maxDstWidth < dstSize[count].width)
+            maxDstWidth = dstSize[count].width;
+        if (minDstHeight > dstSize[count].height)
+            minDstHeight = dstSize[count].height;
+        if (minDstWidth > dstSize[count].width)
+            minDstWidth = dstSize[count].width;
+
+        count++;
+    }
+    closedir(dr1);
+
+    ioBufferSize = (unsigned long long)maxHeight * (unsigned long long)maxWidth * (unsigned long long)ip_channel * (unsigned long long)noOfImages;
+    oBufferSize = (unsigned long long)maxDstHeight * (unsigned long long)maxDstWidth * (unsigned long long)ip_channel * (unsigned long long)noOfImages;
+
+    Rpp8u *input = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *input_second = (Rpp8u *)calloc(ioBufferSize, sizeof(Rpp8u));
+    Rpp8u *output = (Rpp8u *)calloc(oBufferSize, sizeof(Rpp8u));
+
+    Rpp16f *inputf16 = (Rpp16f *)calloc(ioBufferSize, sizeof(Rpp16f));
+    Rpp16f *inputf16_second = (Rpp16f *)calloc(ioBufferSize, sizeof(Rpp16f));
+    Rpp16f *outputf16 = (Rpp16f *)calloc(ioBufferSize, sizeof(Rpp16f));
+
+    Rpp32f *inputf32 = (Rpp32f *)calloc(ioBufferSize, sizeof(Rpp32f));
+    Rpp32f *inputf32_second = (Rpp32f *)calloc(ioBufferSize, sizeof(Rpp32f));
+    Rpp32f *outputf32 = (Rpp32f *)calloc(ioBufferSize, sizeof(Rpp32f));
+
+    Rpp8s *inputi8 = (Rpp8s *)calloc(ioBufferSize, sizeof(Rpp8s));
+    Rpp8s *inputi8_second = (Rpp8s *)calloc(ioBufferSize, sizeof(Rpp8s));
+    Rpp8s *outputi8 = (Rpp8s *)calloc(ioBufferSize, sizeof(Rpp8s));
+
+    RppiSize maxSize, maxDstSize;
+    maxSize.height = maxHeight;
+    maxSize.width = maxWidth;
+    maxDstSize.height = maxDstHeight;
+    maxDstSize.width = maxDstWidth;
+
+    DIR *dr2 = opendir(src);
+    DIR *dr2_second = opendir(src_second);
+    count = 0;
+    i = 0;
+    while ((de = readdir(dr2)) != NULL)
+    {
+        if (strcmp(de->d_name, ".") == 0 || strcmp(de->d_name, "..") == 0)
+            continue;
+
+        count = (unsigned long long)i * (unsigned long long)maxHeight * (unsigned long long)maxWidth * (unsigned long long)ip_channel;
+
+        char temp[1000];
+        strcpy(temp, src1);
+        strcat(temp, de->d_name);
+
+        char temp_second[1000];
+        strcpy(temp_second, src1_second);
+        strcat(temp_second, de->d_name);
+
+        if (ip_channel == 3)
```

Comment:
Codacy found an issue: [The comparison 'ip_channel == 3' is always true.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)